src/polycubed/src/server/Parser/YangNodes.tcpp

```diff
@@ -294,13 +308,59 @@ void Yang::ParseList(
   parent->AddChild(std::move(resource));
 }
 
+/* TODO
+void Yang::ParseAugment(
+            const lys_node_augment *augment, std::queue<std::string> parsed_names,
+            const std::shared_ptr<Resources::Body::ParentResource> &parent,
+            const bool generate_endpoint, ServiceMetadata *md) const {
+        const auto &name_metric = FetchExtensionByIndex(
+                augment->ext,
+                GetExtensionIndex(augment->ext, augment->ext_size, "name-metric", 11));
+        if(!name_metric.empty()) {
+            const auto &type_metric = FetchExtensionByIndex(
+                    augment->ext,
+                    GetExtensionIndex(augment->ext, augment->ext_size, "type-metric", 11));
+            const auto &path_metric = FetchExtensionByIndex(
+                    augment->ext,
+                    GetExtensionIndex(augment->ext, augment->ext_size, "path-metric", 11));
+            const auto &help_metric = FetchExtensionByIndex(
+                    augment->ext,
+                    GetExtensionIndex(augment->ext, augment->ext_size, "help-metric", 11));
+            const auto &type_operation = FetchExtensionByIndex(
+                    augment->ext,
+                    GetExtensionIndex(augment->ext, augment->ext_size, "type-operation", 11));
+            md->Metrics.push_back({name_metric, type_metric, path_metric, help_metric,type_operation});
+
+        }
+    }
+*/
+
 void Yang::ParseLeaf(
     const lys_node_leaf *leaf, std::queue<std::string> parsed_names,
     const std::shared_ptr<Resources::Body::ParentResource> &parent,
-    const bool generate_endpoint) const {
+    const bool generate_endpoint, ServiceMetadata *md) const {
   const auto &cli_example = FetchExtensionByIndex(
       leaf->ext,
       GetExtensionIndex(leaf->ext, leaf->ext_size, "cli-example", 11));
+  const auto &name_metric = FetchExtensionByIndex(
```

Comment:
Add a comment here. Not clear what is this about.

Reply:
Done :+1: